specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2023-08-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,2890 @@
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ReplicaProvisioningState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Snapshot": {
+      "description": "The snapshot resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "The name of the snapshot.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The type of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/SnapshotProperties",
+          "description": "All snapshot properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "SnapshotProperties": {
+      "description": "All snapshot properties.",
+      "type": "object",
+      "required": [
+        "filters"
+      ],
+      "properties": {
+        "provisioningState": {
+          "description": "The provisioning state of the snapshot.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "status": {
+          "description": "The current status of the snapshot.",
+          "enum": [
+            "Provisioning",
+            "Ready",
+            "Archived",
+            "Failed"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "SnapshotStatus",
+            "modelAsString": true
+          }
+        },
+        "filters": {
+          "description": "A list of filters used to filter the key-values included in the snapshot.",
+          "type": "array",
+          "minItems": 1,
+          "maxItems": 3,
+          "items": {
+            "$ref": "#/definitions/KeyValueFilter"
+          },
+          "x-ms-identifiers": []
+        },
+        "compositionType": {
+          "description": "The composition type describes how the key-values within the snapshot are composed. The 'key' composition type ensures there are no two key-values containing the same key. The 'key_label' composition type ensures there are no two key-values containing the same key and label.",
+          "type": "string",
+          "enum": [
+            "Key",
+            "Key_Label"
+          ],
+          "x-ms-enum": {
+            "name": "CompositionType",
+            "modelAsString": true
+          }
+        },
+        "created": {
```

Comment:
>created [](http://example.com/codeflow?start=9&length=7)

This should be replaced by systemData at the top-level, which includes a creation date. Use common types (Resource definition) for top-level properties to get definition for systemData automatically.

Reply:
Pending question in feedback above about whether SystemData needs to be fully filled out if present

Reply:
Answered there.